packages/plugins/graphql/server/services/builders/utils.js

```diff
@@ -25,6 +25,8 @@ module.exports = ({ strapi }) => {
 
       // Components
       if (modelType === 'component') {
+        if (!multiple) return {};
```

Comment:
We could still filter even if single. that would just skip certain components that we don't want

Reply:
That would break the non-null constraint on required non-repeatable component attributes though